txscript/sigcache.go

```diff
@@ -7,18 +7,31 @@ package txscript
 import (
 	"bytes"
 	"crypto/rand"
+	"crypto/sha256"
 	"sync"
 
 	"github.com/btcsuite/btcd/btcec"
 	"github.com/btcsuite/btcd/wire"
 )
 
-// sigInfo represents an entry in the SigCache. Entries in the sigcache are a
-// 3-tuple: (sigHash, sig, pubKey).
-type sigInfo struct {
-	sigHash wire.ShaHash
-	sig     string
-	pubKey  string
+// sigKey represents the lookup key of the signature cache.
+type sigKey [wire.HashSize]byte
+
+// newSigKey creates a new sigcache lookup key using the passed paramters. This
```

Comment:
+// newSigKey creates a new sigcache lookup key using the passed paramters~~parameters~~. This